pkg/jenkinsfile/pipeline.go

```diff
@@ -762,3 +791,206 @@ func (a *CreateJenkinsfileArguments) GenerateJenkinsfile(resolver ImportFileReso
+
+	dir = strings.Replace(dir, util.PlaceHolderAppName, args.GitName, -1)
+	dir = strings.Replace(dir, util.PlaceHolderOrg, args.GitOrg, -1)
+	dir = strings.Replace(dir, util.PlaceHolderGitProvider, args.GitHost, -1)
+	dir = strings.Replace(dir, util.PlaceHolderDockerRegistryOrg, strings.ToLower(args.DockerRegistryOrg), -1)
+
+	if strings.HasPrefix(dir, "./") {
+		dir = args.WorkspaceDir + strings.TrimPrefix(dir, ".")
+	}
+	if !filepath.IsAbs(dir) {
+		dir = filepath.Join(args.WorkspaceDir, dir)
+	}
+
+	if step.GetCommand() != "" {
+		if containerName == "" {
+			containerName = args.DefaultImage
+			log.Logger().Warnf("No 'agent.container' specified in the pipeline configuration so defaulting to use: %s\n", containerName)
+		}
+
+		s := syntax.Step{}
+		args.StepCounter++
+		prefix := prefixPath
+		if prefix != "" {
+			prefix += "-"
+		}
+		stepName := step.Name
+		if stepName == "" {
+			stepName = "step" + strconv.Itoa(1+args.StepCounter)
+		}
+		s.Name = prefix + stepName
+		s.Command = replaceCommandText(step)
+		if args.CustomImage != "" {
+			s.Image = args.CustomImage
+		} else {
+			s.Image = containerName
+		}
+
+		s.Dir = dir
+
+		modifyStep := c.modifyStep(s, dir, args.DockerRegistry, args.DockerRegistryOrg, args.GitName, args.ProjectID, args.KanikoImage, args.NoKaniko)
+
+		steps = append(steps, modifyStep)
+	} else if step.Loop != nil {
+		// Just copy in the loop step without altering it.
+		// TODO: We don't get magic around image resolution etc, but we avoid naming collisions that result otherwise.
+		steps = append(steps, *step)
+	}
+	for _, s := range step.Steps {
+		// TODO add child prefix?
+		childPrefixPath := prefixPath
+		args.WorkspaceDir = dir
+		nestedSteps, nestedCounter := c.CreatePipelineSteps(s, childPrefixPath, args)
+		args.StepCounter = nestedCounter
+		steps = append(steps, nestedSteps...)
+	}
+	return steps, args.StepCounter
+}
+
+// replaceCommandText lets remove any escaped "\$" stuff in the pipeline library
+// and replace any use of the VERSION file with using the VERSION env var
+func replaceCommandText(step *syntax.Step) string {
+	answer := strings.Replace(step.GetFullCommand(), "\\$", "$", -1)
+
+	// lets replace the old way of setting versions
+	answer = strings.Replace(answer, "export VERSION=`cat VERSION` && ", "", 1)
+	answer = strings.Replace(answer, "export VERSION=$PREVIEW_VERSION && ", "", 1)
+
+	for _, text := range []string{"$(cat VERSION)", "$(cat ../VERSION)", "$(cat ../../VERSION)"} {
+		answer = strings.Replace(answer, text, "${VERSION}", -1)
+	}
+	return answer
+}
+
+// modifyStep allows a container step to be modified to do something different
+func (c *PipelineConfig) modifyStep(parsedStep syntax.Step, workspaceDir, dockerRegistry, dockerRegistryOrg, appName, projectID, kanikoImage string, noKaniko bool) syntax.Step {
+	if !noKaniko {
+		if strings.HasPrefix(parsedStep.GetCommand(), "skaffold build") ||
```

Comment:
Yeah, another one I just don't know why it's there. =)

Reply:
Then we are two

Reply:
@jstrachan can you shine some light on what's actually happening here and why?

Reply:
Hello, I know it's alittle bit old piece of code, but it looks like no one was working on it since then :) 
This looks like a crazy hack to use kaniko with serverless setup, I've spent last 3h on finding that out.
Because useKaniko is global settings, someone decided to  replace part of build pack on the fly ... and forgot to document that. Do you think that it make sense to improve documentation or it's better to refactor this part of code (no idea how yet)